presto-main/src/main/java/io/prestosql/server/SessionPropertyDefaults.java

```diff
@@ -106,8 +116,34 @@ public Session newSessionWithDefaultProperties(Session session, Optional<String>
                 queryType,
                 resourceGroupId);
 
-        Map<String, String> systemPropertyOverrides = configurationManager.getSystemSessionProperties(context);
+        Map<String, String> systemPropertyOverrides = filterInvalidSystemPropertyOverrides(configurationManager.getSystemSessionProperties(context));
         Map<String, Map<String, String>> catalogPropertyOverrides = configurationManager.getCatalogSessionProperties(context);
```

Comment:
Catalog-specific session properties should also be validated. There's a `validateCatalogSessionProperty` method for that.

Reply:
@martint I agree that they should be validated as well. `validateCatalogSessionProperty` is invoked when there is a corresponding transaction id. This happens currently in `Session#beginTransactionId` which is invoked at a later point than `newSessionWithDefaultProperties`. If the query already has a transactionId, we can probably validate here, but not sure about the other (and more common) case. 

Should we apply the overrides only after transactionId has been created ?